Carbon.Domain.EntityFrameworkCore.Extensions/EFExtensions.cs

```diff
@@ -303,6 +305,64 @@ public static PagedList<T> ToPagedListEntityFilteredWithSolution<T, U>(
             return result;
         }
 
+        /// <summary>
+        /// Retrieves a paged list of entities with filtered relations based on specified criteria and ordering.
+        /// In this method, data is filtered according to ownership information.
+        /// </summary>
+        /// <typeparam name="T">The type of entities that have ownership relationships with type U and implement IEntity and IHaveOwnership interfaces.</typeparam>
+        /// <typeparam name="U">The type of entity relations that are associated with type T and implement EntitySolutionRelation interface.</typeparam>
+        /// <param name="relationEntities">The IQueryable of relation entities containing pairs of entities of type T and relations of type U.</param>
+        /// <param name="filter">A list of Guids used to filter the relations based on SolutionId, can be null or empty to include all relations.</param>
+        /// <param name="orderables">A list of Orderable objects specifying the ordering of the entities.</param>
+        /// <param name="pageIndex">The index of the page to retrieve (starting from 0).</param>
+        /// <param name="pageSize">The maximum number of entities per page.</param>
+        /// <returns>A PagedList containing the requested page of entities with filtered relations, based on the specified criteria.</returns>
+        /// <remarks>
+        /// This method retrieves a paged list of entities of type T that have ownership relationships with type U.
+        /// The method filters the relations based on the provided list of Guids, which represent SolutionIds, if the filter list is not null or empty.
+        /// The resulting entities are ordered based on the specified ordering criteria and returned as a paged list.
+        /// </remarks>
+        public static async Task<PagedList<T>> ToPagedListEntityFilteredWithSolutionAsync<T, U>(
+            this IQueryable<EFExtensions.RelationEntityPair<T, U>> relationEntities,
+            List<Guid> filter,
+            IList<Orderable> orderables,
+            int pageIndex,
+            int pageSize)
+            where T : IHaveOwnership<U>, IEntity
+            where U : EntitySolutionRelation
+        {
+            var filterContainsData = filter != null && filter.Any();
+
+            var query = relationEntities
+                .Where(k => !filterContainsData || k.Relation == null || filter.Contains(k.Relation.SolutionId));
+
+
+            var filteredQuery = query.OrderBy(orderables);
```

Comment:
@anil-eladag "orderables" must be optional for the query performance. So if can we check orderables is {Count:>0}, it will be useful.